MQ2PluginManager.cpp

```diff
@@ -94,6 +100,78 @@ void DoPluginTool(PSPAWNINFO pChar, PCHAR szLine)
 	}
 }
 
+static MQPlugin* FindMQ2Plugin(const char* szLine)
+{
+	MQPlugin* pPlugin = pPlugins;
+	while (pPlugin)
+	{
+		if (!_stricmp(const_cast<char*>(szLine), pPlugin->szFilename))
+		{
+			return pPlugin;
+		}
+		pPlugin = pPlugin->pNext;
+	}
+	return nullptr;
+}
+
+
+void DrawPluginManager_MQSettingsPanel() {
+	ImGui::TextColored(ImVec4(1.0f, 1.0f, 0.0f, 1.0f), "MQ2PluginManager");
+
+	std::vector<CPluginInfo*> pluginList = PluginTree->GetCurrentPluginList();
+
+	ImGui::SeparatorText("Plugins");
+
+	if (ImGui::BeginTable("PluginTable", 2, ImGuiTableFlags_Borders | ImGuiTableFlags_RowBg)) {
+		int idx = 0;
+		for (auto& plugin : pluginList) {
+			const char* pluginName = plugin->GetName();
+			bool isLoaded = FindMQ2Plugin(pluginName) != nullptr;
+			ImVec4 textColor = isLoaded ? ImVec4(0.0f, 1.0f, 0.0f, 1.0f) : ImVec4(0.867f, 0.410f, 0.510f, 1.0f);
+
+			if (idx == 0) {
+				ImGui::TableNextRow();
+			}
+
+			ImGui::TableSetColumnIndex(idx);
+			ImGui::PushID(pluginName); // Ensure unique ID for each row
+
+			// Combine plugin name and toggle icon into one string
+			std::string combinedLabel = std::string(isLoaded ? ICON_FA_TOGGLE_ON : ICON_FA_TOGGLE_OFF) + "\t" + (pluginName) ;
```

Comment:
This is only used at the end of this function, just move it down and put it into the TextColored call. No need for a std::String